### ISSUE TITLE
Make Vector.shrink not shift contents if srcDim==dstDim

### ISSUE THREAD

**Issue by Flamefire** [ComputationalRadiationPhysics/picongpu#1173]
The Vector.shrink method can be used to make a n-i dimensional vector out of a n dimensional vector.
However if one e.g. wants to have always a 2D vector one would expect that `vec2D.shrink<2>(1)` would not alter the vector. But the values get shifted in every case.

I propose adding a check if srcDim==dstDim and returning the vector unchanged in that case.

**Comment by ax3l** [comment]
I can't follow the full behaviour you are describing, can you add an example pls?
What would `vec2D.shrink<2>(0)` do?

**Comment by Flamefire** [comment]
Yes this is the function i meant.

What I mean is this method is called `shrink`. So you expect that to shrink the vector to the given number of dimensions. Having this cause a shift is at least completely unexpected. Especially in templated code you want to make sure, you have a vector of a given number of dimensions (e.g. a 2D vector out of a 2D or 3D simdim-vector) so you call shrink on it. Causing a rotation in a 2D simulation is vertainly not what you intended.

**Comment by ax3l** [comment]
> you expect that to shrink the vector to the given number of dimensions. Having this cause a shift is at least completely unexpected.

yes that is an unideal mix of functionality, but the problem is: each shrink looses information and people want to choose *which* information (component) shall be lost :)

Isn't `vec2D.shrink<2>(0)` not what you want?
Your proposal would actually introduce a special case to "ignore the offset parameter" in case the dimension stays the same, which makes it imho even more complicated.

**Comment by Flamefire** [comment]
I still think that `shrink<n>` means 'make this vector have n dims'. But yes it ignores the parameter as the full meaning (currently) of `shrink<n>(i)` is 'make this vector have n values from this vector starting at index i'. So probably shrink is an unfortunate name then.

**Comment by ax3l** [comment]
yes thought so too because it basically selects components from an offset round-robin style and creates a new (possibly shrinked, newDim `<=` oldDim) vector
